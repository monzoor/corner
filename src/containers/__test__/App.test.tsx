import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import App from '..';

afterEach(cleanup);

it('renders', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders initial app view', () => {
  render(<App />);
  const linkElement = screen.getByText(/Not playing to anything/i);
  expect(linkElement).toBeInTheDocument();
});

test('Clicking on channel name to play', () => {
  render(<App />);
  const channelListButton = screen.getByTestId('ChannelToPlay1');

  fireEvent.click(channelListButton);

  expect(screen.getByAltText(/open/));
  expect(screen.getByText(/CURRENTLY PLAYING/));
});

test('Clicking on channel to stop', () => {
  render(<App />);
  const channelListButton = screen.getByTestId('ChannelToPlay1');

  fireEvent.click(channelListButton);
  fireEvent.click(channelListButton);

  expect(screen.getByText(/Not playing to anything/));
});
