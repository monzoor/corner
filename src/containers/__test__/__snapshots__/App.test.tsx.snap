// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="wrapper"
    >
      <button
        type="button"
      >
        <img
          alt=""
          height="48px"
          src="/assets/back-arrow.png"
          width="28px"
        />
      </button>
      <p>
        STATIONS
      </p>
      <button
        type="button"
      >
        <img
          alt=""
          height="52px"
          src="/assets/switch.png"
          width="44px"
        />
      </button>
    </div>
    <div
      class="content"
    >
      <ul
        class="wrapper"
      >
        <li
          class="wrapper"
        >
          <div
            class="list"
            data-testid="ChannelToPlay0"
          >
            <p
              class="list--title"
            >
              Putin FM
            </p>
            <p
              class="list--sub-title"
            >
              66,6
            </p>
          </div>
          <div
            class="list__content--hide"
          >
            <div
              class="list__content"
            >
              <button
                type="button"
              >
                <img
                  alt=""
                  src="/assets/minus.png"
                />
              </button>
              <img
                alt="close"
                class="list__content--player"
                src="/assets/play.jpg"
              />
              <button
                type="button"
              >
                <img
                  alt=""
                  src="/assets/plus.png"
                />
              </button>
            </div>
          </div>
        </li>
        <li
          class="wrapper"
        >
          <div
            class="list"
            data-testid="ChannelToPlay1"
          >
            <p
              class="list--title"
            >
              Dribble FM
            </p>
            <p
              class="list--sub-title"
            >
              101,2
            </p>
          </div>
          <div
            class="list__content--hide"
          >
            <div
              class="list__content"
            >
              <button
                type="button"
              >
                <img
                  alt=""
                  src="/assets/minus.png"
                />
              </button>
              <img
                alt="close"
                class="list__content--player"
                src="/assets/play.jpg"
              />
              <button
                type="button"
              >
                <img
                  alt=""
                  src="/assets/plus.png"
                />
              </button>
            </div>
          </div>
        </li>
        <li
          class="wrapper"
        >
          <div
            class="list"
            data-testid="ChannelToPlay2"
          >
            <p
              class="list--title"
            >
              Doge FM
            </p>
            <p
              class="list--sub-title"
            >
              99,4
            </p>
          </div>
          <div
            class="list__content--hide"
          >
            <div
              class="list__content"
            >
              <button
                type="button"
              >
                <img
                  alt=""
                  src="/assets/minus.png"
                />
              </button>
              <img
                alt="close"
                class="list__content--player"
                src="/assets/play.jpg"
              />
              <button
                type="button"
              >
                <img
                  alt=""
                  src="/assets/plus.png"
                />
              </button>
            </div>
          </div>
        </li>
        <li
          class="wrapper"
        >
          <div
            class="list"
            data-testid="ChannelToPlay3"
          >
            <p
              class="list--title"
            >
              Ballads FM
            </p>
            <p
              class="list--sub-title"
            >
              87,1
            </p>
          </div>
          <div
            class="list__content--hide"
          >
            <div
              class="list__content"
            >
              <button
                type="button"
              >
                <img
                  alt=""
                  src="/assets/minus.png"
                />
              </button>
              <img
                alt="close"
                class="list__content--player"
                src="/assets/play.jpg"
              />
              <button
                type="button"
              >
                <img
                  alt=""
                  src="/assets/plus.png"
                />
              </button>
            </div>
          </div>
        </li>
        <li
          class="wrapper"
        >
          <div
            class="list"
            data-testid="ChannelToPlay4"
          >
            <p
              class="list--title"
            >
              Maximum FM
            </p>
            <p
              class="list--sub-title"
            >
              142,2
            </p>
          </div>
          <div
            class="list__content--hide"
          >
            <div
              class="list__content"
            >
              <button
                type="button"
              >
                <img
                  alt=""
                  src="/assets/minus.png"
                />
              </button>
              <img
                alt="close"
                class="list__content--player"
                src="/assets/play.jpg"
              />
              <button
                type="button"
              >
                <img
                  alt=""
                  src="/assets/plus.png"
                />
              </button>
            </div>
          </div>
        </li>
      </ul>
      <div
        class="wrapper"
      >
        <p>
          Not playing to anything
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
